pkg/controllers/provisioning/provisioner.go

```diff
@@ -161,32 +161,34 @@ func (p *Provisioner) getPods(ctx context.Context) ([]*v1.Pod, error) {
 func (p *Provisioner) schedule(ctx context.Context, pods []*v1.Pod) ([]*scheduler.Node, error) {
 	defer metrics.Measure(schedulingDuration.WithLabelValues(injection.GetNamespacedName(ctx).Name))()
 
-	// Get instance type options
-	instanceTypes, err := p.cloudProvider.GetInstanceTypes(ctx)
-	if err != nil {
-		return nil, fmt.Errorf("getting instance types, %w", err)
-	}
-
 	// Build node templates
 	var nodeTemplates []*scheduling.NodeTemplate
 	var provisionerList v1alpha5.ProvisionerList
+	instanceTypes := map[string][]cloudprovider.InstanceType{}
 	if err := p.kubeClient.List(ctx, &provisionerList); err != nil {
 		return nil, fmt.Errorf("listing provisioners, %w", err)
 	}
 	for i := range provisionerList.Items {
-		requirements, err := p.cloudProvider.GetRequirements(ctx, provisionerList.Items[i].Spec.Provider)
+		provisioner := &provisionerList.Items[i]
+		requirements, err := p.cloudProvider.GetRequirements(ctx, provisioner.Spec.Provider)
 		if err != nil {
 			return nil, fmt.Errorf("getting provider requirements, %w", err)
 		}
-		nodeTemplates = append(nodeTemplates, scheduling.NewNodeTemplate(&provisionerList.Items[i], requirements))
+		// Get instance type options
+		instanceType, err := p.cloudProvider.GetInstanceTypes(ctx, provisioner.Spec.Provider)
```

Comment:
```suggestion
		instanceTypes, err := p.cloudProvider.GetInstanceTypes(ctx, provisioner.Spec.Provider)
```

Reply:
changed this to InstanceTypeOptions as we are appending this to a dictionary later in the code.